pkg/kubelet/kubelet_network.go

```diff
@@ -89,6 +91,115 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	return supported
 }
 
+func wouldFit(kl *Kubelet, pod *v1.Pod, composedSearch []string, domain string) (bool, []string) {
+	searchLen := len(strings.Join(composedSearch, " "))
+
+	if searchLen == 0 || (searchLen+len(domain)+1 <= types.MaxSearchLineLen && len(composedSearch) < types.MaxSearchLineDNSDomains) {
+		return true, append(composedSearch, domain)
+	} else {
+		log := fmt.Sprintf("Search Line limits were exceeded, some dns names have been omitted, the applied search line is: %s", strings.Join(composedSearch, " "))
+		kl.recorder.Event(pod, v1.EventTypeWarning, "DNSSearchForming", log)
+		glog.Error(log)
+	}
+	return false, composedSearch
+}
+
+func (kl *Kubelet) formDNSSearchForDNSDefault(hostSearch []string, pod *v1.Pod) []string {
+	if len(hostSearch) <= types.MaxSearchLineDNSDomains && len(strings.Join(hostSearch, " ")) <= types.MaxSearchLineLen {
```

Comment:
re-worked dns search line forming logic, at first just exclude all duplicates using map, after cut it to fit limits if needed.